.github/workflows/libzmq.yml

```diff
@@ -0,0 +1,59 @@
+name: libzmq
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+    paths:
+      - '.github/workflows/libzmq.yml'
+      - 'ports/libzmq/**'
+  pull_request:
+    branches:
+      - main
+    paths:
+      - '.github/workflows/libzmq.yml'
+      - 'ports/libzmq/**'
+
+jobs:
+  build:
+    runs-on: self-hosted
+    if: |
+      (github.event_name == 'workflow_dispatch') ||
+      (github.event_name == 'push' && github.ref == 'refs/heads/main') ||
+      (github.event_name == 'pull_request' && contains(github.event.pull_request.title, 'libzmq'))
+    steps:
+    - name: Checkout build-files
+      uses: actions/checkout@v4
+      with:
+        path: build-files
+
+    - name: Checkout libzmq
+      uses: actions/checkout@v4
+      with:
+        repository: zeromq/libzmq
+        path: libzmq
+        ref: v4.3.5
+
+    - name: Login to GitHub Container Registry
+      uses: docker/login-action@v3
+      with:
+        registry: ghcr.io
+        username: ${{github.actor}}
+        password: ${{secrets.GITHUB_TOKEN}}
+
+    - name: Pull build environment
+      run: |
+        docker pull ghcr.io/qnx-ports/sdp800-build-env:latest
+
+    - name: Build libzmq
+      uses: addnab/docker-run-action@v3
+      with:
+        image: ghcr.io/qnx-ports/sdp800-build-env:latest
+        options: -v ${{ github.workspace }}:/home/runner/workspace -e MAKEFLAGS=${{ env.MAKEFLAGS }}
+        shell: bash
+        run: |
+          source ~/qnx/800/qnxsdp-env.sh
+          cd workspace
+          make -C libzmq/build_qnx JLEVEL=${nproc} install
```

Comment:
`JLEVEL=${nproc}` shouldn't be needed as `options` now includes MAKEFLAGS, which allocated available cores on the machines accordingly.